### PR TITLE
Move duplicate logic in unit test to helper function

### LINKED ISSUE #79: Move duplicate logic in unit test to helper function
## Why
There are a lot of duplicate code in unit test such as
```go
err = json.Unmarshal(respBodyData, &parsedRespBody)
if err != nil {
	log.Error(errorHelper.JSONUnmarshalFailure, err)
}
```
We can simplify this by moving this logic to another function and reuse it anywhere.

### Acceptance Criteria
- Simplify unit test to DRY by moving duplicate logic to helper function.

### PR DESCRIPTION
**Pull request by gutakk**
## What happened
Resolve #79 
- Move duplicate logic in the unit test to test helper function.

**Note**
Project structure for test and test helper will be done in another task.

## Insight
N/A

## Proof Of Work
- All tests must be passed.

### CODE REVIEW

**Comment by gutakk** [comment]
@carryall
I think it is better to keep `error_constant`. So I fixed https://github.com/gutakk/go-google-scraper/pull/78#discussion_r573428080 https://github.com/gutakk/go-google-scraper/pull/78#discussion_r573428178 and https://github.com/gutakk/go-google-scraper/pull/78#discussion_r573421573 in this PR.

File: controllers/api_v1/login_test.go [hunk 1]
```diff
@@ -1,66 +1,39 @@
...
 )
 
 func init() {
 	gin.SetMode(gin.TestMode)
 
-	err := os.Chdir(path_test.GetRoot())
-	if err != nil {
-		log.Fatal(errorconf.ChangeToRootDirFailure, err)
-	}
+	path_test.ChangeToRootDir()
```
Comment (olivierobert) [inline controllers/api_v1/login_test.go]: It's quite inconsistent with other renamed packages just as `testDB`, `testHttp`, `testjson`. So how about renaming all test packages on import:

```suggestion
	testOauth "github.com/gutakk/go-google-scraper/tests/oauth_test"
	testPath "github.com/gutakk/go-google-scraper/tests/path_test"
```

File: controllers/api_v1/login_test.go [hunk 2]
```diff
@@ -1,66 +1,39 @@
...
-	"github.com/gutakk/go-google-scraper/controllers/api_v1"
 	"github.com/gutakk/go-google-scraper/db"
-	"github.com/gutakk/go-google-scraper/helpers/log"
 	"github.com/gutakk/go-google-scraper/models"
-	"github.com/gutakk/go-google-scraper/oauth"
 	testConfig "github.com/gutakk/go-google-scraper/tests/config"
 	testDB "github.com/gutakk/go-google-scraper/tests/db"
+	"github.com/gutakk/go-google-scraper/tests/fabricator"
 	testHttp "github.com/gutakk/go-google-scraper/tests/http"
+	testjson "github.com/gutakk/go-google-scraper/tests/json"
```
Comment (olivierobert) [inline controllers/api_v1/login_test.go]: To be consistent, the casing should be:

```suggestion
	testJson "github.com/gutakk/go-google-scraper/tests/json"
```

**Review by olivierobert — COMMENTED** [review]
Overall, it is a major improvement. I am still unsure about all the sub-packages for tests e.g. `testPath`, `testJson`... As in the Gin template, I do think that having all of these helpers in the same test package would be better.

File: tests/http/http.go [hunk 3]
```diff
@@ -33,3 +37,57 @@ func perform(req *http.Request, r http.Handler, headers http.Header) *httptest.R
...
+	readCloserBody, isReadCloser := respBody.(io.ReadCloser)
+
+	if isByteBuffer {
+		bodyByte, err = ioutil.ReadAll(byteBufferBody)
+	} else if isReadCloser {
+		bodyByte, err = ioutil.ReadAll(readCloserBody)
+	}
+
+	if err != nil {
+		log.Error(errorconf.ReadResponseBodyFailure, err)
```
Comment (carryall) [inline tests/http/http.go]: @gutakk Does `log.Error` fail the test? If no, then do you think we should make the test failed?

**Comment by gutakk** [comment]
> Overall, it is a major improvement. I am still unsure about all the sub-packages for tests e.g. `testPath`, `testJson`... As in the Gin template, I do think that having all of these helpers in the same test package would be better.

I plan to refactor this in https://github.com/gutakk/go-google-scraper/issues/36

File: controllers/api_v1/login_test.go (hunk 1 above)
Comment (gutakk) [reply controllers/api_v1/login_test.go]: Updated testOauth in e0cdbf4ea90321b3a8bfbee1d6d516fb8113e7dd
Updated testPath in a862983dfbb5f93e113b4bef792b140f9be5be34

File: controllers/api_v1/login_test.go (hunk 2 above)
Comment (gutakk) [reply controllers/api_v1/login_test.go]: Updated testJson in 47c1b0be9ed95166672e085e19c38521a3fffc21 89adea6421a3e626bbcd26c0104d0a340021f15d

File: tests/http/http.go (hunk 3 above)
Comment (gutakk) [reply tests/http/http.go]: Good catch!. `log.Error` won't fail the test.
Updated to use fatal instead in c087a2d570a3b6a6e604195f15866d1806453119

**Review by olivierobert — APPROVED** [review]

**Review by carryall — APPROVED** [review]

**Pull request merged**